test/publishers/idp_rule_processing_publisher_test.py

```diff
@@ -0,0 +1,227 @@
+        },
+        {
+            "path": "bucket/path/to/archive.tgz",
+            "metadata": {
+                "cluster_id": "uuid",
+                "external_organization": "an organization"
+            },
+            "report": {
+                "reports": [],
+            },
+        },
+        id="adding optional elements",
+    ),
+]
+
+
+@pytest.mark.parametrize("input, expected_output", VALID_INPUT_MSG)
+def test_publish_valid(input, expected_output):
+    """Check that Kafka producer is called with an expected message."""
+    report = '{"reports": []}'
+
+    expected_output = json.dumps(expected_output) + "\n"
+    sut = IDPRuleProcessingPublisher("outgoing_topic", {"bootstrap.servers": "kafka:9092"})
+    sut.producer = MagicMock()
+
+    sut.publish(input, report)
+    sut.producer.produce.assert_called_with("outgoing_topic", expected_output.encode())
+
+
+INVALID_REPORTS = [
+    None,
+    1,
+    2.0,
+    1 + 3j,
+    [],
+    {},
+    "",
+]
+
+
+@pytest.mark.parametrize("invalid_report", INVALID_REPORTS)
+def test_publish_invalid_report(invalid_report):
+    """Check the behaviour of publish when an invalid report is received."""
+    sut = IDPRuleProcessingPublisher("outgoing_topic", {"bootstrap.servers": "kafka:9092"})
+    sut.producer = MagicMock()
+
+    with pytest.raises(CCXMessagingError):
+        sut.publish(VALID_INPUT_MSG, invalid_report)
+        assert not sut.producer.produce.called
+
+
+@pytest.mark.parametrize("input,output", VALID_INPUT_MSG)
+def test_error(input, output):
+    """Check that error just prints a log."""
+    _ = output  # output values are not needed
+
+    sut = IDPRuleProcessingPublisher("outgoing_topic", {"bootstrap.servers": "kafka:9092"})
+
+    with patch("ccx_messaging.publishers.idp_rule_processing_publisher.log") as log_mock:
+        sut.error(input, None)
+        assert log_mock.warning.called
+
+
+VALID_REPORTS = [
+    pytest.param(
+        json.dumps(
+            {"reports": [], "pass": [], "info": [], "workload_recommendations": []}
+        ),
+        {
+            "path": "bucket/path/to/archive.tgz",
+            "metadata": {
+                "cluster_id": "uuid",
+            },
+            "report": {"reports": [], "pass": [], "info": [], "workload_recommendations": []},
+        },
+        id="valid_report",
+    ),
+    pytest.param(
+        json.dumps(
+            {"reports": [], "pass": [], "info": [], "workload_recommendations": []}
+        ),
+        {
+            "path": "bucket/path/to/archive.tgz",
+            "metadata": {
+                "cluster_id": "uuid",
+            },
+            "report": {"reports": [], "pass": [], "info": [], "workload_recommendations": []},
+        },
+        id="valid_report",
+    )
+]
+
+
+@pytest.mark.parametrize("input,expected_output", VALID_REPORTS)
+def test_filter_ocp_rules_results(input, expected_output):
+    """Check that the workload recommendations are filtered out from the engine results."""
```

Comment:
don't the `input` and `expected_output` both contain the `workload_recommendation` field? is that expected? I thought we were dropping that one after filtering

Reply:
There is no filtering out of the workload_recommendations key in the internal data pipeline. It is being filtered out in the external one.

Reply:
if there is no filtering the unit test docstring is wrong then:
> """Check that the workload recommendations are filtered out from the engine results."""

Reply:
Oh, you are right, and probably the whole test doesn't make sense in this context